### PR TITLE
chore: upgrade to polkadot-v0.9.17

### PR DESCRIPTION
**Pull request by Roznovjak**
Fixes: #399

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "basilisk"
-version = "5.3.1"
+version = "5.3.2"
```
Comment (lumir-mrkva) [inline node/Cargo.toml:3]:
```suggestion
version = "5.4.0"
```

**Comment by lumir-mrkva** [comment]
looks like some changes in github api are causing there CI failures

**Comment by green-jay** [comment]
I had a look and LGTM when the build passes

**Review by lumir-mrkva — CHANGES_REQUESTED** [review]

File: node/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "basilisk"
-version = "5.4.0"
+version = "5.4.1"
```
[the remark this answers is not in the document]
Comment (Roznovjak) [reply node/Cargo.toml]: done

**Review by martinfridrich — APPROVED** [review]
Looks good to me.

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ on:
   pull_request:
 
 env:
-  RELAY_CHAIN_VERSION: v0.9.16
+  RELAY_CHAIN_VERSION: v0.9.17-rc4
```
Comment (green-jay) [inline .github/workflows/workflow.yml:12]: why not just 0.9.17?

Comment (lumir-mrkva) [reply .github/workflows/workflow.yml:12]: there is no release on tag v0.9.17, this is official release https://github.com/paritytech/polkadot/releases/tag/v0.9.17-rc4

**Review by lumir-mrkva — APPROVED** [review]

**Pull request merged**